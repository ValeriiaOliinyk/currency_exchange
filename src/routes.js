export default {
  home: "/",
  currency: "/currency",
};
