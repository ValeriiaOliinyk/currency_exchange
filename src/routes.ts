export default {
  home: "/",
  currency: "/currency",
  chart: "/chart"
};
