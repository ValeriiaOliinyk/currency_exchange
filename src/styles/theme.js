export default {
  colors: {
    main: "#a6dbfb",
    text: "#0e2b3c",
    dark: "#56809b",
    hover: "#67a030",
    white: "#ffffff",
  },
};
