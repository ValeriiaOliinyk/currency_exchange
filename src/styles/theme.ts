export default {
  colors: {
    main: "#a6dbfb",
    text: "#0e2b3c",
    dark: "#56809b",
    hover: "#67a030",
    white: "#ffffff",
    red: "#800000",
    black: "#000000",
    transparent: "#ffffff00"
  },

  breakpoints: {
    mobileL: `screen and (max-width: 425px)`,
    tablet: `screen and (max-width: 768px)`,
    tabletFrom: `screen and (min-width: 768px)`
  }
};
