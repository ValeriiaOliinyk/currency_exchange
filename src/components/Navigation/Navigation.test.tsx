import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import "jest-styled-components";
import routes from "../../routes";

// Components
import {
  Header,
  Nav,
  NavigationList,
  NavigationItems,
  Logo,
  NavLinks,
} from "../../styled";
import theme from "../../styles/theme";

it("Navigation component renders correctly", () => {
  const tree = renderer
    .create(
      <Header>
        <BrowserRouter>
          <Nav>
            <Logo theme={theme} />
            <NavigationList theme={theme}>
              <NavigationItems>
                <NavLinks to={routes.home} theme={theme}>
                  Home
                </NavLinks>
              </NavigationItems>
              <NavigationItems>
                <NavLinks to={routes.currency} theme={theme}>
                  Convert
                </NavLinks>
              </NavigationItems>
            </NavigationList>
          </Nav>
        </BrowserRouter>
      </Header>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
