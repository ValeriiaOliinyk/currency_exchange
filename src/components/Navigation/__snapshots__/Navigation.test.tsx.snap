// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders correctly 1`] = `
.c0 {
  width: 100%;
  height: 80px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0px 0px 5px 0px rgba(0,0,0,0.75);
}

.c2 {
  width: 45px;
  height: 45px;
  background-image: url(return-of-investment.svg);
  background-repeat: no-repeat;
  background-size: cover;
}

.c1 {
  width: 80%;
  height: 70%;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  font-size: 24px;
  color: #a6dbfb;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  -webkit-transition: 250ms color linear;
  transition: 250ms color linear;
}

.c4:hover {
  color: #67a030;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 300px;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

@media screen and (max-width:768px) {
  .c2 {
    width: 35px;
    height: 35px;
  }
}

@media screen and (max-width:768px) {
  .c4 {
    font-size: 14px;
  }
}

@media screen and (max-width:768px) {
  .c3 {
    width: 200px;
  }
}

<header
  className="c0"
>
  <nav
    className="c1"
  >
    <div
      className="c2"
    />
    <ul
      className="c3"
    >
      <li
        className=""
      >
        <a
          aria-current="page"
          className="c4 active"
          href="/"
          onClick={[Function]}
          style={Object {}}
          theme={
            Object {
              "breakpoints": Object {
                "mobileL": "screen and (max-width: 425px)",
                "tablet": "screen and (max-width: 768px)",
              },
              "colors": Object {
                "black": "#000000",
                "dark": "#56809b",
                "hover": "#67a030",
                "main": "#a6dbfb",
                "red": "#800000",
                "text": "#0e2b3c",
                "transparent": "#ffffff00",
                "white": "#ffffff",
              },
            }
          }
        >
          Home
        </a>
      </li>
      <li
        className=""
      >
        <a
          aria-current={null}
          className="c4"
          href="/currency"
          onClick={[Function]}
          theme={
            Object {
              "breakpoints": Object {
                "mobileL": "screen and (max-width: 425px)",
                "tablet": "screen and (max-width: 768px)",
              },
              "colors": Object {
                "black": "#000000",
                "dark": "#56809b",
                "hover": "#67a030",
                "main": "#a6dbfb",
                "red": "#800000",
                "text": "#0e2b3c",
                "transparent": "#ffffff00",
                "white": "#ffffff",
              },
            }
          }
        >
          Convert
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
